File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.c
int* smallerNumbersThanCurrent(int* nums, int numsSize, int* returnSize){
    int *k;
    int i,j,cnt;
    k = (int*)malloc(sizeof(int) * numsSize);
    
    for(i = 0; i < numsSize; ++i)
    {
        cnt = 0;
        for(j = 0; j < numsSize; ++j)
        {
            if(i != j && (nums[i] > nums[j]))
                cnt++;
        }
        k[i] = cnt;
    }
    
    *returnSize = numsSize;
    return k;
}

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] freq = new int[101];

        for(int i : nums)
            freq[i]++;
        for(int i = 1; i < 101; i++)
            freq[i] += freq[i-1];
        
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] != 0)
                nums[i] = freq[nums[i] - 1];
        }
        return nums;
    }
}